pkg/chain/bitcoin/electrs_test.go

```diff
@@ -90,3 +90,117 @@ func TestElectrsConnection_vbyteFee(t *testing.T) {
+			"invalid character ',' after object key",
+		},
+		"unexpected integer response": {
+			"bcrt1qy6n80gen875en87ka798svvzrneq2erhhwfzzf",
+			200,
+			"73",
+			"json: cannot unmarshal number into Go value of type []interface {}",
+		},
+	}
+	apiURL := "example.org/api"
+	for testName, testData := range testData {
+		t.Run(testName, func(t *testing.T) {
+			electrs := NewElectrsConnection(apiURL)
+			electrs.setClient(mockClient{
+				mockGet: func(url string) (*http.Response, error) {
+					expectedURL := fmt.Sprintf("%s/address/%s/txs", apiURL, testData.btcAddress)
+					if url != expectedURL {
+						t.Errorf("unexpected url\nexpected: %s\nactual:   %s", expectedURL, url)
+					}
+					return &http.Response{
+						StatusCode: testData.responseCode,
+						Body:       ioutil.NopCloser(bytes.NewReader([]byte(testData.response))),
+					}, nil
+				},
+				mockPost: func(url string, contentType string, reader io.Reader) (*http.Response, error) {
+					return nil, nil
+				},
+			})
```

Comment:
This seems to be shared across the tests, maybe we could extract it to a separate funcion?